Crash while adding network acls

So as not to be spammy, I've attached as a gist:

https://gist.github.com/sethlindberg/68311cfe488636995bb3

the issue happened when applying a self-created module that applies a set of network acls to subnets. If you need more information I'm happy to give it.

Upgrading left a more interesting error message:

```
2015/04/17 11:14:59 [DEBUG] root.acls_private_a: eval: *terraform.EvalApply
2015/04/17 11:14:59 [DEBUG] apply: aws_network_acl.main: executing Apply
2015/04/17 11:14:59 terraform-provider-aws: 2015/04/17 11:14:59 [DEBUG] Network Acl create config: &ec2.CreateNetworkACLRequest{DryRun:(aws.BooleanValue)(nil), VPCID:(aws.StringValue)(0xc2085ab3c0)}
2015/04/17 11:14:59 terraform-provider-aws: 2015/04/17 11:14:59 [INFO] Network Acl ID: acl-bfc147da
2015/04/17 11:15:02 terraform-provider-aws: panic: runtime error: index out of range
2015/04/17 11:15:02 terraform-provider-aws: 
2015/04/17 11:15:02 terraform-provider-aws: goroutine 215 [running]:

<snip stacktrace>

2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalWriteState
2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalApplyProvisioners
2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalIf
2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalWriteDiff
2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalIf
2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalWriteState
2015/04/17 11:15:02 [DEBUG] root.acls_private_a: eval: *terraform.EvalApplyPost
2015/04/17 11:15:02 [ERROR] root.acls_private_a: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* unexpected EOF
2015/04/17 11:15:02 [ERROR] root.acls_private_a: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* unexpected EOF
2015/04/17 11:15:02 [ERROR] root.acls_private_a: eval: *terraform.EvalOpFilter, err: 1 error(s) occurred:

* unexpected EOF
2015/04/17 11:15:02 [ERROR] root.acls_private_a: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* unexpected EOF
2015/04/17 11:15:02 waiting for all plugin processes to complete...
2015/04/17 11:15:02 [DEBUG] /usr/local/bin/terraform-provider-aws: plugin process exited
```

The configuration snippet I posted above was wrapped in a module, which may be different from how you tested. I dunno, I'm at a loss for words, here.  It almost looks like it is applying the acl configuration in an infinite loop.

--- a/builtin/providers/aws/resource_aws_network_acl.go
+++ b/builtin/providers/aws/resource_aws_network_acl.go
@@ -185,7 +185,6 @@ func resourceAwsNetworkAclUpdate(d *schema.ResourceData, meta interface{}) error
 	}
 
 	if d.HasChange("subnet_id") {
-
 		//associate new subnet with the acl.
 		_, n := d.GetChange("subnet_id")
 		newSubnet := n.(string)
@@ -355,9 +354,11 @@ func findNetworkAclAssociation(subnetId string, conn *ec2.EC2) (networkAclAssoci
 	if err != nil {
 		return nil, err
 	}
-	for _, association := range resp.NetworkACLs[0].Associations {
-		if *association.SubnetID == subnetId {
-			return association, nil
+	if resp.NetworkACLs != nil && len(resp.NetworkACLs) > 0 {
+		for _, association := range resp.NetworkACLs[0].Associations {
+			if *association.SubnetID == subnetId {
+				return association, nil
+			}
 		}
 	}
 	return nil, fmt.Errorf("could not find association for subnet %s ", subnetId)

